Automattic/jetpack
# User-less connection: Restrict first connection to admins only

This PR restricts the first user connection to admins only and hides the corresponding prompts (banners, buttons etc) from the UI when a non-admin user visits the Jetpack dashboard and there is only a site-level connection (user-less).

This issue surfaced while we are working with User-less connections. Before, the admin user that set up Jetpack had to also connect their WPCOM account, so any secondary non-admin users were eligible to connect their account afterwards.

With user-less any non-admin user could try to connect their account, causing the following error:
![Screenshot 2021-02-09 at 4 12 01 PM](https://user-images.githubusercontent.com/1758399/107385012-2d0b8f80-6afb-11eb-99a7-e2b8cd288720.png)


#### Changes proposed in this Pull Request:

* Connection: Introduces a new method, `current_user_can_connect_account` in `Manager` class that checks if the current user is an administrator or a connected owner already exists
* Client/UI: Utilizes the above method, to pass on this information to the client and based on that hides any banners/links/buttons that prompt the user to connect their account if they are not allowed.

#### Jetpack product discussion


p1HpG7-aJm-p2

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:
- Create a new JN site and make sure user-less mode is enabled ( Settings -> Jetpack Constants -> JETPACK_NO_USER_TEST_MODE)
- Add various non-admin users, with roles: editor, author, subscriber, contributor
- Connect your Jetpack Site, **skipping user authorization**. Now you have User-less Jetpack setup.
- Login as each of the users you created, making sure that you don't see any prompts to link your account. More specifically:
* You shouldn't see the banner to link your account at the top of the Jetpack dashboard.
* You shouldn't see the User connection section under Connections (at-a-glance) at all.
* You shouldn't see the button to connect your account to view more stats (at-a-glance) at the Stats section.
- Login as admin again and make sure that you see all the above.
- **Connect your admin user** , log out and login a last time as eg author
- Verify that now the prompts to link your account **are visible again**.

#### Proposed changelog entry for your changes:


*n/a::Ensuring existing connect flows don't break for user-less connections.
